### DISCUSSION TITLE
Error: @prisma/client did not initialize yet

### DISCUSSION THREAD

**Discussion by franciscop** [prisma/prisma#10149] [Q&A]
Following [the Quickstarter](https://www.prisma.io/docs/getting-started/quickstart) but changing it to plain ESM Javascript instead of Typescript I find this error:

> /Users/francisco/projects/starter/node_modules/.prisma/client/index.js:3
    throw new Error(
> Error: @prisma/client did not initialize yet. Please run "prisma generate" and try to import it again.
> In case this error is unexpected for you, please report it in https://github.com/prisma/prisma/issues

Notes about similar threads I've seen around:
- I don't have a global prisma installed, so `prisma generate` => `zsh: command not found: prisma`
- Node version `v15.7.0`
- I added `"type": "module"` to my package.json since I'm using ESM with import/export
- Changed my start script to be `"node ./script.js"` since I'm using ESM with import/export
- Changed the code a bit since it seems it couldn't import a named dep. Now my code is as follows:

```js
import pkg from "@prisma/client";

const { PrismaClient } = pkg;
const prisma = new PrismaClient();
const allUsers = await prisma.user.findMany({
  include: { posts: true }
});
console.log(allUsers);
await prisma.$disconnect();
```

Note that it's throwing on the `import pkg from "@prisma/client";`

**Comment by franciscop** [comment]
Following [this blog post](https://flaviocopes.com/prisma-fix-initialize-yet-vercel/) was solved, adding the `postinstall` script:

```
"postinstall": "prisma generate"
```

So I suggest adding it to the official example so that it works.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
